week2/RO_flowsheet_build_verbose.ipynb

```diff
@@ -0,0 +1,369 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.fs.feed = Feed(property_package=m.fs.properties)\n",
+    "\n",
+    "m.fs.pump = Pump(\n",
+    "    property_package=m.fs.properties,\n",
+    ")\n",
+    "\n",
+    "m.fs.RO = ReverseOsmosis0D(\n",
+    "    property_package=m.fs.properties,\n",
+    "    has_pressure_change=True,\n",
+    "    pressure_change_type=PressureChangeType.calculated,\n",
+    "    mass_transfer_coefficient=MassTransferCoefficient.calculated,\n",
+    "    concentration_polarization_type=ConcentrationPolarizationType.calculated,\n",
+    ")\n",
+    "\n",
+    "m.fs.product = Product(property_package=m.fs.properties)\n",
+    "m.fs.brine = Product(property_package=m.fs.properties)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2f09f602",
+   "metadata": {},
+   "source": [
+    "### Step 2e: Define Unit Model Connectivity\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step2e.png\" width=\"600\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "e6e252d3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.fs.feed_to_pump = Arc(source=m.fs.feed.outlet, destination=m.fs.pump.inlet)\n",
+    "m.fs.pump_to_RO = Arc(source=m.fs.pump.outlet, destination=m.fs.RO.inlet)\n",
+    "m.fs.RO_to_product = Arc(source=m.fs.RO.permeate, destination=m.fs.product.inlet)\n",
+    "m.fs.RO_to_brine = Arc(source=m.fs.RO.retentate, destination=m.fs.brine.inlet)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "84719040",
+   "metadata": {},
+   "source": [
+    "### Step 2f: Expand Arcs\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step2f.png\" width=\"600\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "96faa069",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TransformationFactory(\"network.expand_arcs\").apply_to(m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4250afc",
+   "metadata": {},
+   "source": [
+    "### Step 2g: Add Variables, Constraints, and Objectives\n",
+    "\n",
+    "<center><img src=\"graphics/workflow-step2g.png\" width=\"600\" /></center>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "60036788",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "m.fs.RO.flux_LMH = Var(\n",
+    "    initialize=20.0,\n",
+    "    bounds=(0, 45),\n",
+    "    units=pyunits.liter / (pyunits.m**2 * pyunits.hr),\n",
+    "    doc=\"Water flux through the membrane in LMH\",\n",
+    ")\n",
+    "\n",
+    "m.fs.RO.flux_LMH_constraint = Constraint(\n",
+    "    expr=m.fs.RO.flux_LMH\n",
+    "    == pyunits.convert(\n",
+    "        m.fs.RO.mixed_permeate[0].flow_vol_phase[\"Liq\"] / m.fs.RO.area,\n",
+    "        to_units=pyunits.liter / (pyunits.m**2 * pyunits.hr),\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "m.fs.obj = Objective(expr=m.fs.RO.flux_LMH, sense=-1)  # Maximize flux"
```

Comment:
interesting, but we actually aren't maximizing flux with a simulation, right? And if we did unfix DOF at the end, what would we unfix for a meaningful result? Would flux not just go to upper bound, say if unfixing pressure, recovery? 

In any case, I might lean towards removing this, unless you have them do a try it yourself at the end, where they unfix recovery and pressure, solve again, then display flux, pressure, recovery.